addons/website/models/website.py

```diff
@@ -1267,3 +1267,77 @@ def _get_cached_values(self):
 
     def _get_cached(self, field):
         return self._get_cached_values()[field]
+
+    def _get_snippets_containers(self):
+        html_containers = [('ir_ui_view', 'arch_db')]
+        self.env.cr.execute(r"""
+            SELECT model,
+                   name
+            FROM ir_model_fields
+            WHERE ttype = 'html'
+            AND store = 't'
+        """)
+        html_fields = self.env.cr.fetchall()
+        for model, name in html_fields:
+            orm_model = self.env[model]
+            if not orm_model._abstract:
+                html_containers.append((orm_model._table, name))
+        return html_containers
+
+    def _get_snippets_assets(self):
+        """Returns every snippet asset from the database in the form of a list :
+        [snippet_module, snippet_id, asset_version, asset_type, asset_id]
+        """
+        self.env.cr.execute(r"""
+            SELECT regexp_matches[1] AS snippet_module,
+                   regexp_matches[2] AS snippet_id,
+                   regexp_matches[3] AS asset_version,
+                   CASE WHEN regexp_matches[4]='scss' THEN 'css'
+                        ELSE regexp_matches[4]
+                   END AS asset_type,
+                   id AS asset_id
+            FROM
+                (SELECT REGEXP_MATCHES(PATH, '(\w*)\/.*\/snippets\/(\w*)\/(\d{3})\.(js|scss)'),
+                        id
+                FROM ir_asset) AS regexp;
+        """)
+        return self.env.cr.fetchall()
+
+    def _get_snippet_occurences(self, snippet_module, snippet_id, asset_version, asset_type, html_containers):
+        snippet_occurences = []
+        # We check every snippet template definition
+        snippet_template = self.env.ref('%s.%s' % (snippet_module, snippet_id), raise_if_not_found=False)
+        if snippet_template:
+            snippet_template_html = snippet_template._render().decode()
+            match = re.search('<([^>]*class="[^>]*)>', snippet_template_html)
+            if match:
+                snippet_occurences.append(match.group())
+
+        # As well as every snippet dropped in a potential container
+        snippet_regex = '<([^>]*data-snippet="%s"[^>]*)>' % snippet_id
+        snippet_dropped = "UNION ".join("""SELECT REGEXP_MATCHES(%s, '%s') FROM %s """ % (column, snippet_regex, table) for table, column in html_containers)
+        self.env.cr.execute(snippet_dropped)
+        results = self.env.cr.fetchall()
+        for r in results:
+            snippet_occurences.append(r[0][0])
+
+        count = 0
```

Comment:
Why a counter? Just use a boolean and break the loop if you find one
It will be faster and the result will be the same as you actually just want to know if there is at least one, but knowing how much is useless